private/buf/cmd/buf/buf.go

```diff
@@ -150,23 +148,36 @@ func NewRootCommand(name string) *appcmd.Command {
 				Use:   "mod",
 				Short: "Configure and update buf modules.",
 				SubCommands: []*appcmd.Command{
-					modinit.NewCommand("init", builder, "", false),
+					modinit.NewCommand("init", builder, modInitDeprecationMessage, true), // TODO(alex): Think about this one (see TODO below).
+					export.NewCommand("export", builder, "", false),                      // TODO(alex): Move this command into the modexport package.
 					modprune.NewCommand("prune", builder),
+					push.NewCommand("push", builder, "", false), // TODO(alex): Move this command into the modpush package.
 					modupdate.NewCommand("update", builder, "", false),
 					modclearcache.NewCommand("clear-cache", builder, "", false, "cc"),
 				},
 			},
+			// TODO(alex): We might actually want to keep "buf mod init" in the "mod" sub-command since it's directly tied to
+			// the other "mod" commands. For example, I initialize a module with "buf mod init", then push it immediately after
+			// with "buf mod push". Otherwise, we would need to jump between multiple sub-commands (initialize with "config",
+			// then push with "mod").
+			//
+			// However, keeping it in mod is awkward for "config ls-{breaking,lint}-rules" - listing the content of the `buf.yaml`
+			// should exist alongside how it's initialized.
+			//
+			// How do we foresee future configuration commands for `buf.work.yaml` and `buf.gen.yaml` to look? Will these be new
+			// sub-commands under the "config" sub-command? What does that suggest about how we should structure this (i.e. "buf config mod-init")?
```

Comment:
This is related to the comment above - we eventually want `buf config init` to be more interactive and support bootstrapping the user's entire `buf` filesystem layout. For example, providing a wizard that asks the user where their `.proto` roots are, and creating a `buf.work.yaml`, `buf.yaml`, and `buf.gen.yaml` in the appropriate places.